### PR TITLE
Refactor ChunkAppend plan selection

### PR DESCRIPTION
**Pull request by svenklemm**
Move ChunkAppend selection of next subplan into separate function

### CODE REVIEW

File: src/chunk_append/exec.c [hunk 1]
```diff
@@ -333,18 +333,44 @@ chunk_append_exec(CustomScanState *node)
...
+		state->current = NO_MATCHING_SUBPLANS;
+		return;
+	}
+
+	if (state->runtime_exclusion)
+	{
+		if (!state->runtime_initialized)
+			initialize_runtime_exclusion(state);
+
+		if (bms_is_empty(state->valid_subplans))
```
Comment (JLockerman) [inline src/chunk_append/exec.c]: how come we do this now even if `state->runtime_initialized`?

Comment (svenklemm) [reply src/chunk_append/exec.c]: runtime exclusion might have excluded all but bms_next_member should come to the same conclusion so this check can be removed

File: src/chunk_append/exec.c [hunk 2]
```diff
@@ -333,18 +333,41 @@ chunk_append_exec(CustomScanState *node)
 		 * more subplans, return the empty slot set up for us by
```
Comment (cevian) [inline src/chunk_append/exec.c]: the part of the comment about returning empty slot is now out of date? (happens above?)

File: src/chunk_append/exec.c [hunk 3]
```diff
@@ -293,9 +293,9 @@ chunk_append_exec(CustomScanState *node)
 
 		CHECK_FOR_INTERRUPTS();
 
-		/*
-		 * figure out which subplan we are currently processing
-		 */
+		if (state->current == NO_MATCHING_SUBPLANS)
+			return ExecClearTuple(node->ss.ps.ps_ResultTupleSlot);
+
```
Comment (cevian) [inline src/chunk_append/exec.c]: can we Assert(state->current != INVALID_SUBPLAN_INDEX) here?

**Pull request merged**